viscy/representation/embedding_writer.py

```diff
@@ -44,47 +59,198 @@ def _move_and_stack_embeddings(
+                _, UMAP = _fit_transform_umap(features, **default_UMAP_kwargs)
+                for i in range(UMAP.shape[1]):
+                    ultrack_indices[f"UMAP{i+1}"] = UMAP[:, i]
+            except Exception as e:
+                _logger.warning(f"UMAP computation failed: {str(e)}")
+
+        # Compute PHATE if requested
+        if "PHATE" in reductions:
+            try:
+                _logger.debug("Computing PHATE")
+                _, PHATE = compute_phate(features, **default_PHATE_kwargs)
+                for i in range(PHATE.shape[1]):
+                    ultrack_indices[f"PHATE{i+1}"] = PHATE[:, i]
+            except Exception as e:
+                _logger.warning(f"PHATE computation failed: {str(e)}")
+
+        # Compute PCA if requested
+        if "PCA" in reductions:
+            try:
+                _logger.debug("Computing PCA")
+                PCA_features, _ = compute_pca(features, **default_PCA_kwargs)
+                for i in range(PCA_features.shape[1]):
+                    ultrack_indices[f"PCA{i+1}"] = PCA_features[:, i]
+            except Exception as e:
+                _logger.warning(f"PCA computation failed: {str(e)}")
+
+    # Create multi-index and dataset
+    index = pd.MultiIndex.from_frame(ultrack_indices)
+
+    # Create dataset dictionary with features
+    dataset_dict = {"features": (("sample", "features"), features)}
+
+    # Add projections if provided
+    if projections is not None:
+        dataset_dict["projections"] = (("sample", "projections"), projections)
+
+    # Create the dataset
+    dataset = Dataset(dataset_dict, coords={"sample": index}).reset_index("sample")
+
+    _logger.debug(f"Writing dataset to {output_path}")
+    with dataset.to_zarr(output_path, mode="w") as zarr_store:
+        zarr_store.close()
+
+
+class EmbeddingWriter(BasePredictionWriter):
+    """
+    Callback to write embeddings to a zarr store in an Xarray-compatible format.
+
+    Parameters
+    ----------
+    output_path : Path
+        Path to the zarr store.
+    write_interval : Literal["batch", "epoch", "batch_and_epoch"], optional
+        When to write the embeddings, by default 'epoch'.
+    PHATE_kwargs : dict, optional
+        Keyword arguments passed to PHATE, by default None.
+    UMAP_kwargs : dict, optional
+        Keyword arguments passed to UMAP, by default None.
+    PCA_kwargs : dict, optional
+        Keyword arguments passed to PCA, by default None.
+    reductions : List[Literal["PHATE", "UMAP", "PCA"]], optional
+        List of dimensionality reduction methods to compute, by default all available methods.
+    """
+
+    def __init__(
+        self,
+        output_path: Path,
+        write_interval: Literal["batch", "epoch", "batch_and_epoch"] = "epoch",
+        PHATE_kwargs: dict | None = None,
+        UMAP_kwargs: dict | None = None,
+        PCA_kwargs: dict | None = None,
+        reductions: list[Literal["PHATE", "UMAP", "PCA"]] | None = ["PHATE", "PCA"],
```

Comment:
the reductions might be redundant. perhaps we just do PHATE_kwargs and PCA_kwargs. If they are None then compute them?